### PR TITLE
ci: add codeql scanning

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR adds automatic vulnerability scanning

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'javascript', 'python' ]
```
Comment (daniel-heppner-ibigroup) [inline .github/workflows/codeql-analysis.yml:37]: why are we checking python?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "dev", master ]
```
Comment (daniel-heppner-ibigroup) [inline .github/workflows/codeql-analysis.yml:18]: what's going on with the quotes?

**Review by daniel-heppner-ibigroup — APPROVED** [review]

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (miles-grant-ibigroup) [reply .github/workflows/codeql-analysis.yml:37]: datatools does include some python! https://github.com/ibi-group/datatools-ui/blob/4e7b8eac69fd14ae40b8a13ea86c3bf8b5e44691/scripts/loadLegacy.py

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (miles-grant-ibigroup) [reply .github/workflows/codeql-analysis.yml:18]: I'm not sure, but this file is generated by GitHub so I trust them!

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "dev" ]
+  schedule:
+    - cron: '24 19 * * 6'
```
Comment (binh-dam-ibigroup) [inline .github/workflows/codeql-analysis.yml:23]: Is this scan intended to be periodical even on PRs to `dev`?

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, just wondering about whether the periodicity is needed on top of the PR-based runs.

File: .github/workflows/codeql-analysis.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (binh-dam-ibigroup) [inline .github/workflows/codeql-analysis.yml:3]: Add a comment saying the file was initially generated by github.

**Pull request merged**